rootex/framework/entity_factory.cpp

```diff
@@ -141,19 +141,18 @@ Ref<Component> EntityFactory::createDefaultComponent(const String& name)
 	}
 }
 
-Ref<Entity> EntityFactory::createEntity(TextResourceFile* entityJSONDescription, bool isEditorOnly)
+Ref<Entity> EntityFactory::createEntity(const JSON::json& entityJSON, const String& filePath, bool isEditorOnly)
```

Comment:
Isn't `TextResourceFile` supposed to encapsulate both these information along with caching?
What was the need to split that parameter into these?

Reply:
createEntity is reused in createEntityfromClass for which create Entity will need a json argument

Reply:
We need to handle creating from custom JSON objects, so it got break, BUT a wrapper function [here](https://github.com/sdslabs/Rootex/pull/226/files#diff-a556ba660ee5bad8ce95f374d9059248R41) that does the original createEntity(TextResourceFile*) version is good to have for interfacing cleanly with Lua and C++.

Just add a new createEntity() which takes in a text resource file and uses the json overloaded function internally